controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -384,9 +387,24 @@ public CompletableFuture<CreateStreamStatus> createStream(String scope, String s
             return CompletableFuture.completedFuture(
                     CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_STREAM_NAME).build());
         }
+        try {
+            validateStreamConfig(streamConfig);
+        } catch (IllegalArgumentException e) {
+            log.error(requestId, "Create stream: {} failed due to invalid retention policy {}", stream, streamConfig);
+            return CompletableFuture.completedFuture(
+                    CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_RETENTION_POLICY).build());
+        }
         return callCreateStream(scope, stream, streamConfig, createTimestamp, requestId);
     }
 
+    @VisibleForTesting
+    protected void validateStreamConfig(StreamConfiguration streamConfig) {
+        boolean isInvalidStreamConfig =  Config.REQUIRE_RETENTION_POLICY
+                && !streamConfig.getScalingPolicy().equals(ScalingPolicy.fixed(1))
```

Comment:
can streamConfig.getScalingPolicy() be null ?

Reply:
> can streamConfig.getScalingPolicy() be null ?

From StreamConfigurationBuilder I could see that by default Scalling policy is `ScalingPolicy.fixed(1)`; So as per me it cant be null.

Reply:
We have a null check already in place for scaling in `ControllerIMPL.java` so even if the user passes null, the system won't allow